commit 4695c42e1f6e4ca55ede54d558f83ebcf0c2288b (refs/changes/47/1647/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-10T19:26:03+00:00 (2 years, 7 months ago)
    
    DOCKER-1013 broke purpose-less certificates in DOCKER-1005
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/auth.js b/lib/auth.js
index 5ab41d7..68fb2c7 100644
--- a/lib/auth.js
+++ b/lib/auth.js
@@ -73,7 +73,7 @@ function authTls(req, res, next) {
      * the expiry time and use the issuer's CN to look up the real account
      * key.
      */
-    if (cert.purposes.indexOf('joyentDocker') !== -1) {
+    if (cert.purposes && cert.purposes.indexOf('joyentDocker') !== -1) {
         log.trace('found "joyentDocker" certificate purpose, will treat as '
             + 'new-style certificate');
 
